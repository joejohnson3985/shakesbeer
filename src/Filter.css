.filter {
  width: 200px;
  height: 30px;
  border: 1px solid darkorange;
  font-size: 16px;
  margin-right: 30px;
}

.filter-label {
  padding-right: 10px;
}

.filter:focus {
  outline-color: #EB8240;
}