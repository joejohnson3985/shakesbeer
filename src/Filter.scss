@import './variables.scss';

.filter {
  width: 200px;
  height: 30px;
  border: 1px solid $primary-color;
  font-size: 16px;
}

.filter-label {
  font-size: 18px;
}

.filter:focus {
  outline-color: $secondary-color;
}