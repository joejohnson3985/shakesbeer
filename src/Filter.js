import React, { Component } from 'react';
import './Filter.css';

class Filter extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    return (
      <div class="filter-container">
        <label htmlFor={this.props.filterName} className='filter-label'>{this.props.label}</label>
        <select className='filter' id={this.props.filterName} onChange={this.props.updateSelected}>
        <option>All</option>
          {
            this.props.filterOptions.map(loc => {
              return(
                <option className='state-dropdown-options' key={loc}>{loc}</option>
              )
            })
          }
        </select>
      </div>
    )
  }
}

export default Filter;