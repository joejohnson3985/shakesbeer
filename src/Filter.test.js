import React from 'react';
import Filter from './Filter.js';
import { shallow } from 'enzyme';

const mockData = [1,2,3,4]
const mockFunc = jest.fn()

describe('Filter', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow (
            <Filter filterName="string" 
                    label="string" 
                    filterOptions={mockData} 
                    updateSelected={mockFunc} 
            />
        )
    });

    it('Should have a snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});