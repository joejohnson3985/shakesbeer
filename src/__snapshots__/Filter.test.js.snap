// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Should match the snapshot 1`] = `
<div>
  <label
    className="filter-label"
  />
  <select
    className="filter"
    onChange={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
  >
    <option>
      All
    </option>
    <option
      className="state-dropdown-options"
      key="1"
    >
      1
    </option>
    <option
      className="state-dropdown-options"
      key="2"
    >
      2
    </option>
    <option
      className="state-dropdown-options"
      key="3"
    >
      3
    </option>
    <option
      className="state-dropdown-options"
      key="4"
    >
      4
    </option>
  </select>
</div>
`;
